src/services/selfHosted/useSelfHostedHasAdmins.ts

```diff
@@ -0,0 +1,58 @@
+import { useQuery } from '@tanstack/react-query'
+import { z } from 'zod'
+
+import Api from 'shared/api'
+import { NetworkErrorObject } from 'shared/api/helpers'
+
+export const HasAdminsSchema = z.object({
+  config: z
+    .object({
+      hasAdmins: z.boolean().nullable(),
+    })
+    .nullable(),
+})
+
+function fetchHasAdmins({ provider }: { provider: string }) {
+  const query = `
+  query HasAdmins {
+    config {
+      hasAdmins
+    }
+  }
+  `
+  return Api.graphql({
+    provider,
+    query,
+  })
+}
+export const useSelfHostedHasAdmins = (
+  { provider }: { provider: string },
+  options = {}
+) => {
+  const opts = {
+    select: ({ data }: { data: z.infer<typeof HasAdminsSchema> }) =>
+      data?.config?.hasAdmins,
+    keepPreviousData: true,
+    ...options,
+  }
+  return useQuery({
+    queryKey: ['hasAdmins', provider],
+    queryFn: () =>
+      fetchHasAdmins({
+        provider,
+      }).then((res) => {
+        const parsedRes = HasAdminsSchema.safeParse(res?.data)
+
+        if (!parsedRes.success) {
+          return Promise.reject({
+            status: 404,
+            data: {},
+            dev: 'useSelfHostedHasAdmins - 404 schema parsing failed',
+          } satisfies NetworkErrorObject)
+        }
+
+        return parsedRes
```

Comment:
👀

Reply:
Another instance of `select` manipulating the `data`. I changed it to look like other places in our app.

updated to use the rejectNetworkErrorHelper too